src/Encoder.c

```diff
@@ -0,0 +1,4344 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+
+***************************************************************************************************/
+
+#include <Zycore/LibC.h>
+#include <Zydis/Encoder.h>
+#include <Zydis/Internal/EncoderData.h>
+#include <Zydis/Internal/SharedData.h>
+
+/* ============================================================================================== */
+/* Macros                                                                                         */
+/* ============================================================================================== */
+
+/* ---------------------------------------------------------------------------------------------- */
+/* Constants                                                                                      */
+/* ---------------------------------------------------------------------------------------------- */
+
+#define ZYDIS_OPSIZE_MAP_BYTEOP    1
+#define ZYDIS_OPSIZE_MAP_DEFAULT64 4
+#define ZYDIS_OPSIZE_MAP_FORCE64   5
+
+/* ---------------------------------------------------------------------------------------------- */
+
+/* ============================================================================================== */
+/* Internal enums and types                                                                       */
+/* ============================================================================================== */
+
+/**
+ * Defines the `ZydisEncoderRexType` enum.
+ */
+typedef enum ZydisEncoderRexType_
+{
+    ZYDIS_REX_TYPE_UNKNOWN,
+    ZYDIS_REX_TYPE_REQUIRED,
+    ZYDIS_REX_TYPE_FORBIDDEN,
+
+    /**
+     * Maximum value of this enum.
+     */
+    ZYDIS_REX_TYPE_MAX_VALUE = ZYDIS_REX_TYPE_FORBIDDEN,
+    /**
+     * The minimum number of bits required to represent all values of this enum.
+     */
+    ZYDIS_REX_TYPE_REQUIRED_BITS = ZYAN_BITS_TO_REPRESENT(ZYDIS_REX_TYPE_MAX_VALUE)
+} ZydisEncoderRexType;
+
+/**
+ * Defines the `ZydisEncodableAttribute` enum.
+ */
+typedef enum ZydisEncoderInstructionAttribute_
```

Comment:
It seems that only `EVEX_B` is not supported by `ZYDIS_ATTRIB_`, so after adding that one it would be a rather simple change. I initially went with my own type because I expected more mismatches with `ZYDIS_ATTRIB_` but it turned out we have only one. It's a promising change.

I guess you decided to use defines for attributes because values exceeded range for 32-bit integers. At least that's how I explained it to myself when I first saw this :P

Reply:
@mappzor is correct. Enums are int sized in C, so they can even be less than 32 bits on some platforms. I think we tried to always use defines for flags.